src/sparseml/transformers/compression/utils/compress_save.py

```diff
@@ -67,16 +71,23 @@ def save_pretrained_wrapper(
             :param sparsity_config: optional sparsity config to compress model with,
             if no config is provided it will be inferred from the model
             :param save_compresed: whether or not to compress the model on disk
+            :param skip_compression_stats: whether to skip the calculation of
+            compression statistics (such as global sparsity and sparsity structure) when
+            saving a model in dense format
             :param kwargs: additional kwargs to pass on to model.save_pretrained
             """
             model = model_ref()
 
             if sparsity_config is not None:
-                # if a sparsity config is provided, always save compressed
                 sparsity_config.fill_config_details(model)
-                save_compressed = True
-            elif save_compressed:
+            elif not skip_compression_stats:
                 # try to infer a sparsity config from the model if none is provided
+                _LOGGER.warning(
```

Comment:
Maybe this should just be an "info" level, since it will run by default all the time